topo_processor/util/aws_files.py

```diff
@@ -0,0 +1,37 @@
+import json
+from urllib.parse import urlparse
+
+import boto3
+from linz_logger import get_log
+
+from topo_processor.util.aws_credentials import assumed_role_session
+
+
+def s3_download(source_path: str, dest_path: str, role_arn: str):
+    get_log().debug("s3_download", objectPath=source_path, destinationPath=dest_path, roleArn=role_arn)
+    url_o = urlparse(source_path)
+    bucket_name = url_o.netloc
+    object_name = url_o.path[1:]
+
+    session = assumed_role_session(role_arn)
+    s3 = session.resource("s3")
+    try:
+        s3.Bucket(bucket_name).download_file(object_name, dest_path)
+    except Exception as e:
+        get_log().error(f"An error has occurred while downloading {source_path}: {e}")
```

Comment:
should log these as key value pairs

Reply:
`get_log().error("An error has occurred during the download.", file=source_path, error=e)` ?